oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/core/AbstractOAuth2AuthorizationServerMetadata.java

```diff
@@ -138,6 +138,16 @@ public B jwkSetUrl(String jwkSetUrl) {
 			return claim(OAuth2AuthorizationServerMetadataClaimNames.JWKS_URI, jwkSetUrl);
 		}
 
+		/**
+		 * Use this {@code userinfo_endpoint} in the resulting {@link AbstractOAuth2AuthorizationServerMetadata}, OPTIONAL.
+		 *
+		 * @param userInfoEndpoint the {@code URL} of the OAuth 2.0 UserInfo Endpoint
+		 * @return the {@link AbstractBuilder} for further configuration
+		 */
+		public B userInfoEndpoint(String userInfoEndpoint) {
```

Comment:
Related to supporting only `oidc-configuration`, please move this to `OidcProviderConfiguration` and update the javadoc accordingly.